### ISSUE TITLE
Missing `state_referenced_locally` compiler warning

### ISSUE THREAD

**Issue by chbert** [pheuter/svelte-check-rs#63]
## Description

`svelte-check-rs` does not report the `state_referenced_locally` compiler warning that the official `svelte-check` reports. This warning is important for Svelte 5 projects using runes, as it catches a common reactivity mistake.

## Minimal Reproduction

Create a file `Test.svelte`:

```svelte
<script lang="ts">
  let { data } = $props();
  
  // This should warn: "This reference only captures the initial value of `data`"
  const { form } = data;
  
  // This should also warn
  let items = $state(data.items);
</script>

<p>Form: {form}</p>
```

### Expected (svelte-check output):
```
Test.svelte:5:19
Warn: This reference only captures the initial value of `data`. Did you mean to reference it inside a closure instead?
https://svelte.dev/e/state_referenced_locally (svelte)

  const { form } = data;
```

### Actual (svelte-check-rs output):
```
svelte-check-rs found 0 errors and 0 warnings
```

## Environment

- svelte-check-rs: 0.6.2
- svelte-check: 4.3.5  
- svelte: 5.x

## Additional Context

This warning was expanded in Svelte 5.45.3 (see [sveltejs/svelte#17289](https://github.com/sveltejs/svelte/issues/17289)) and is documented at https://svelte.dev/e/state_referenced_locally

The warning helps catch cases where reactive state is captured at initialization time rather than staying reactive - a common mistake when migrating to Svelte 5 runes.

In a real project with 222 files, `svelte-check` reports 486 `state_referenced_locally` warnings while `svelte-check-rs` reports 0.
